ws/ShardManager.ts

```diff
@@ -0,0 +1,35 @@
+import DiscordWebsocket, { ConnectionProperties } from './DiscordWebsocket';
+
+interface BotGateway {
+  session_start_limit: {
+    max_concurrency: number;
+    remaining: number;
+    reset_after: number;
+    total: number;
+  };
+  shards: number;
+  url: string;
+}
+
+export default class ShardManager {
+  private shards: Map<number, DiscordWebsocket> = new Map();
+
+  constructor(private token: string, private intents: number, private connProps: Partial<ConnectionProperties> = {}) {
```

Comment:
Why are all these marked private? Also you're not doing anything with it

Reply:
I am using them in line 30. Also they're marked as private because they won't be used anywhere outside of the class.

Reply:
Doesn't matter if they're being used outside of the class or not, still shouldn't be marked as private

Reply:
May I ask why though?

Reply:
Also unless I've misunderstood TS, you still need to assign them to `this` for it to actually work

Reply:
Once you've initialized them with a modifier (e.g., `private`, `public`, `protected`) they'll be assigned to `this`.

Reply:
> 
> 
> Doesn't matter if they're being used outside of the class or not, still shouldn't be marked as private

Isn't the private modifier used for marking properties and functions that should be accessible only within the class though? Since those initialization properties aren't going to be referenced outside of the shard manager, why make them public?

Reply:
We may not use them internally but (as mentioned in other thread) people may want to access them for whatever reason